src/network/p2p/event_loop.rs

```diff
@@ -304,47 +304,58 @@ impl EventLoop {
 						},
 						_ => {},
 					},
-					_ => {},
 				}
 			},
-			SwarmEvent::Behaviour(BehaviourEvent::Identify(event)) => {
-				match event {
-					identify::Event::Received {
-						peer_id,
-						info:
-							Info {
-								listen_addrs,
-								agent_version,
-								protocol_version,
-								..
-							},
-					} => {
-						trace!("Identity Received from: {peer_id:?} on listen address: {listen_addrs:?}");
-						if agent_version != self.identity_data.agent_version
-							&& protocol_version != self.identity_data.protocol_version
+			SwarmEvent::Behaviour(BehaviourEvent::Identify(event)) => match event {
+				identify::Event::Received {
+					peer_id,
+					info:
+						Info {
+							listen_addrs,
+							agent_version,
+							protocol_version,
+							..
+						},
+				} => {
+					trace!(
+						"Identity Received from: {peer_id:?} on listen address: {listen_addrs:?}"
+					);
+					let incoming_peer_agent_version = match AgentVersion::from_str(&agent_version) {
```

Comment:
use let else here?

Reply:
We need both values in this case, so a match is probably a better choice...